### DISCUSSION TITLE
Tree editor with XMI source

### DISCUSSION THREAD

**Discussion by julen9** [eclipse-emfcloud/emfcloud#219] [Theia Tree Editor]
Hi,

I have tested the `tree-editor` extension, but it creates a `.tree` file in **JSON** format, but the problem is I am modeling on another file, this file is a **XMI** format (similar to XML). 

I need the `tree-editor` extension reading and editing from that **XMI** file, because I dont want to maintain 2 different files (model in XMI and the .tree file in JSON) 

`coffee-editor` example it works fine with his `tree-editor`, but I cant see the code diferences. 
Please could you help me?

Thanks in advance,
Julen.

**Comment by ndoschek** [comment]
Hi @julen9,
the main difference is the model management in these two examples.

- The `theia-tree-editor` example simply operates on a JSON file directly.
- The `coffee-editor` example has one general model management component, i.e. the [`Model Server`](https://github.com/eclipse-emfcloud/emfcloud-modelserver). This component is responsible to maintain models in a specified format, in this case XML (but it could of course supports other formats as JSON).
In this example, the tree-editor converts its changes into commands that are sent to the Model Server, which applies them and updates the model file.

So in your case, depending on your final use case, I'd say you have several possibilities to go forward:
- If you only need the tree-editor to operate on XML files directly, you could customize the resource handling and load and save models to an XML file (e.g. by using some external library). Here is a pointer to the [load and save functions in the ResourceTreeEditorWidget](https://github.com/eclipse-emfcloud/theia-tree-editor/blob/master/theia-tree-editor/src/browser/resource/resource-tree-editor-widget.ts#L82).
- If you expect additional editing capabilities in the future, the same structure as the `coffee-editor` could be interesting, then also the model file handling would be centralized and you could use XMI out of box so to say.

HTH and best regards,
Nina
